### ISSUE TITLE
Segmentation fault crash (monster evolution setting?)

### ISSUE THREAD

**Issue by Ymedronic** [CleverRaven/Cataclysm-DDA#27627]
**Describe the bug**  
A sudden crash happening while exploring.

CRASH LOG FILE: config/crash.log
VERSION: 0.C-36464-g60240b8
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:
    @0x5CD4B5[cataclysm-tiles.exe+0x1CD4B5]
    @0x5CE062[cataclysm-tiles.exe+0x1CE062]
    SMPEG_error+0x4B034@0xE64F40[cataclysm-tiles.exe+0xA64F40]
    _C_specific_handler+0x96@0x7FFA58211E26[ntdll.dll+0x81E26]
    _chkstk+0x9D@0x7FFA5822349D[ntdll.dll+0x9349D]
    RtlRaiseException+0xF67@0x7FFA581E48D7[ntdll.dll+0x548D7]
    KiUserExceptionDispatcher+0x3A@0x7FFA5822262A[ntdll.dll+0x9262A]
    @0x8A8B3B[cataclysm-tiles.exe+0x4A8B3B]
    @0x8A9237[cataclysm-tiles.exe+0x4A9237]
    @0x8B0E21[cataclysm-tiles.exe+0x4B0E21]
    @0xD4BC8F[cataclysm-tiles.exe+0x94BC8F]
    @0xD4A47C[cataclysm-tiles.exe+0x94A47C]
    @0xD4B635[cataclysm-tiles.exe+0x94B635]
    @0x8F0932[cataclysm-tiles.exe+0x4F0932]
    @0x906C70[cataclysm-tiles.exe+0x506C70]
    @0x9071F6[cataclysm-tiles.exe+0x5071F6]
    @0xD69A3D[cataclysm-tiles.exe+0x969A3D]
    @0x93A8C6[cataclysm-tiles.exe+0x53A8C6]
    @0x8F12BD[cataclysm-tiles.exe+0x4F12BD]
    @0x907690[cataclysm-tiles.exe+0x507690]
    @0x92C2C7[cataclysm-tiles.exe+0x52C2C7]
    @0x8B5E60[cataclysm-tiles.exe+0x4B5E60]
    @0x8B6108[cataclysm-tiles.exe+0x4B6108]
    @0x6EF64D[cataclysm-tiles.exe+0x2EF64D]
    @0x898DF2[cataclysm-tiles.exe+0x498DF2]
    @0x89B050[cataclysm-tiles.exe+0x49B050]
    IMG_LoadWEBP_RW+0x4CC5B0@0x1443BF0[cataclysm-tiles.exe+0x1043BF0]
    @0x4013ED[cataclysm-tiles.exe+0x13ED]
    @0x4014FB[cataclysm-tiles.exe+0x14FB]
    BaseThreadInitThunk+0x22@0x7FFA56F313D2[KERNEL32.DLL+0x13D2]
    RtlUserThreadStart+0x34@0x7FFA581A54F4[ntdll.dll+0x154F4]

**To Reproduce**  
Create new world, change monster evolution from 4 to 50 or 25 (other values yet to test)
Start near an area with lots of different enemies?
Start exploring/fleeing the enemies, possibly.
crash?

**Expected behavior**  
Not crashing.

**Versions and configuration(please complete the following information):**
 - OS: Windows 8.1
 - Game Version: build 8420
 - Graphics version: Tiles
 - Mods loaded: DDA, disable npc needs, filthy clothing, simplified nutrition, Bens GF recipes, makeshift items, medieval n historic, more survival tools, more locations, fuji's more buildings, more city locations, oa's additional buildings, urban development, boats, folding parts, vehicle additions, tanks and other vehicles, LUA mods (fun, items, skills, traits) Statsthoughskills IV

**Additional context**  
The same mods were used with the default monster evolution settings, and this crash didn't occur.
Sorry if i'm not thorough enough.

**Comment by Ymedronic** [comment]
Update- testing again, 4 monster evolution still produced a crash. :(
It seems this crash started when I updated the game from version 8419 or so. (at least I didn't get this crash previously.) Sorry for misidentifying probable cause.

CRASH LOG FILE: config/crash.log
VERSION: 0.C-36464-g60240b8
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:
    @0x5CD4B5[cataclysm-tiles.exe+0x1CD4B5]
    @0x5CE062[cataclysm-tiles.exe+0x1CE062]
    SMPEG_error+0x4B034@0xE64F40[cataclysm-tiles.exe+0xA64F40]
    _C_specific_handler+0x96@0x7FFA58211E26[ntdll.dll+0x81E26]
    _chkstk+0x9D@0x7FFA5822349D[ntdll.dll+0x9349D]
    RtlRaiseException+0xF67@0x7FFA581E48D7[ntdll.dll+0x548D7]
    KiUserExceptionDispatcher+0x3A@0x7FFA5822262A[ntdll.dll+0x9262A]
    @0xD6CEAB[cataclysm-tiles.exe+0x96CEAB]
    @0xD6E2EE[cataclysm-tiles.exe+0x96E2EE]
    @0x8B411E[cataclysm-tiles.exe+0x4B411E]
    @0x6F3F68[cataclysm-tiles.exe+0x2F3F68]
    IMG_LoadWEBP_RW+0x4CC1A9@0x14437E9[cataclysm-tiles.exe+0x10437E9]
    @0x4013ED[cataclysm-tiles.exe+0x13ED]
    @0x4014FB[cataclysm-tiles.exe+0x14FB]
    BaseThreadInitThunk+0x22@0x7FFA56F313D2[KERNEL32.DLL+0x13D2]
    RtlUserThreadStart+0x34@0x7FFA581A54F4[ntdll.dll+0x154F4]

**Comment by Shodan14** [comment]
Got the same problem after #8417, just rolled back for now.

**Comment by GearheadLydia** [comment]
Duplicate of #27600.
